homeassistant/components/screenlogic/__init__.py

```diff
@@ -125,7 +134,19 @@ async def _async_update_data(self):
             async with self.api_lock:
                 await self.hass.async_add_executor_job(self.gateway.update)
         except ScreenLogicError as error:
-            raise UpdateFailed(error) from error
+            _LOGGER.warning("ScreenLogicError - attempting reconnect: %s", error)
+
+            connect_info = get_connect_info(self.hass, self.config_entry)
+            screen_logic_reconnect = partial(ScreenLogicGateway, **connect_info)
+            try:
+                async with self.api_lock:
+                    self.gateway = await self.hass.async_add_executor_job(
+                        screen_logic_reconnect
+                    )
+                    await self.hass.async_add_executor_job(self.gateway.update)
```

Comment:
It would be even better if we wrapped both sync calls inside a function that we schedule once on the executor. It's expensive to switch context. The function can return the updated gateway.

Reply:
I've abstracted this into a function and now using the executor to run the sync function.  I had to move some stuff around so it would work outside of the `ScreenlogicDataUpdateCoordinator` class, I've tested the changes for several days with no issues.